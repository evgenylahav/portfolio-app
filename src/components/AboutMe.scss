.aboutme-grid{
    grid-template-rows: 1fr 1fr;
}

.aboutme-grid h2{
    font-family: 'Playfair Display', serif;
    font-size: 50px;
    font-weight: bold;
    color: black;
    text-align: center;
    width: 75%;
}

.aboutme-grid h3{
    font-family: 'Indie Flower', cursive;
    font-size: 30px;
    color: black;
    font-weight: bold;
    text-align: center;
    width: 75%;
}
.aboutme hr {
    border-top: 3px solid blue;
    width: 50%;
    margin: auto;
}

.aboutme-grid p{
    font-family: 'Indie Flower', cursive;
    font-size: 20px;
    color: black;
    text-align: center;
    width: 75%;
}