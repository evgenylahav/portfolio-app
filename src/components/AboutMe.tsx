import * as React from 'react';

export interface AboutMeState {

}

export class AboutMe extends React.Component<{}, AboutMeState> {
    constructor(props: any) {
        super(props);
        this.state = {};
    }

    public render() {
        return (
            <h1>About Me</h1>
        );
    }
}