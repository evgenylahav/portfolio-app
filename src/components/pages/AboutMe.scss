.about-header-image {
    width: 100%;
    margin-top: -2em;
    margin-bottom: 2em;
    max-height: 500px;
}
  
.about-profile-pic {
    float: right;
    margin: 0 50px 0 50px;
    width: 20%;
    height: auto;
}

.aboutme-grid{
    grid-template-rows: 1fr 1fr;
}

.aboutme-grid h2{
    font-family: 'Playfair Display', serif;
    font-size: 50px;
    font-weight: bold;
    color: black;
    padding-top: 40px;
    // text-align: center;
    // width: 75%;
}

.aboutme-grid h3{
    font-family: 'Indie Flower', cursive;
    font-size: 30px;
    color: black;
    font-weight: bold;
    text-align: center;
    width: 75%;
}
.aboutme hr {
    border-top: 3px solid blue;
    width: 50%;
    margin: auto;
}

.aboutme-grid p{
    // font-family: 'Indie Flower', cursive;
    // font-size: 20px;
    color: black;
    // text-align: center;
    // width: 75%;
}