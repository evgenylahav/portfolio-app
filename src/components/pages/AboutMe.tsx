import * as React from 'react';
import { Container, Col, Image } from 'react-bootstrap';
import { ImageGallery } from '../common/ImageGallery';
import './AboutMe.scss';
import profilePic from '../../assets/Image-1.png';
import family from '../../assets/family.jpeg';
import programmingPic from '../../assets/programming.jpg';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'


export interface AboutMeProps {}
export interface AboutMeState {}

export class AboutMe extends React.Component<AboutMeProps, AboutMeState> {
    constructor(props: any) {
        super(props);
        this.state = {};
    }

    public render() {
        return (
            <div className='aboutme-grid'>
                <h3>Discover who am I as a person outside of my day job</h3>            
                <Container>
                    <Col xs={12} sm={12} sm-offset={2} style={{paddingBottom: '100px'}}>
                        <Image src={profilePic} roundedCircle className="about-profile-pic" />
                        <h2>A few facts about myself</h2>
                        <div>
                            <FontAwesomeIcon icon="plane-departure"/>
                            <p style={{display: "inline-block"}}>
                                I love traveling. Especially with my family. We have explored several unforgetable places and hopefully many more
                                will follow.
                            </p>
                        </div>
                        <div>
                            <FontAwesomeIcon icon="basketball-ball"/>
                            <p style={{display: "inline-block"}}>
                                I am a sports enthusiast, although, unfortunatelly, as a viewer. I love Tennis and Basketball the most, 
                                but during big events such as the Olympics or the World Cup, I watch everything there is.
                            </p>
                        </div>
                        <div>
                            <FontAwesomeIcon icon="hamburger"/>
                            <p>
                                I love food. Both making and eating. This is really a weakness of mine, and I'm paying for that with a horrible
                                BMI rate.
                            </p>
                        </div>
                        <div>
                            <FontAwesomeIcon icon="coffee"/>
                            <p>
                                I love coffee - can't live without it. I usually drink 5-6 cups a day, with a mixture of cappucinno,
                                black Turkish and my favorite - double espresso.
                            </p>
                        </div>
                        <div>
                            <FontAwesomeIcon icon="tv"/>
                            <p>
                                I am a fan of TV shows, watching quite a few on Netflix and other platforms. Among the best I've
                                watched are: 'Better Call Soul', 'Dark', 'Bron', 'Breaking Bad', 'Fargo', 'Narcos' and others
                            </p>
                        </div>
                         
                        <Image src={family} roundedCircle className="about-profile-pic" />
                        <h2>My family</h2>
                        <div>
                            <FontAwesomeIcon icon="venus"/>
                            <p>
                                I am married to Yardena, a school councelor and a pschycoterapist, and one of the 
                                smartest people I know. 
                            </p>
                        </div>
                        <div>
                            <FontAwesomeIcon icon="child"/>
                            <p>
                                I am a father of Romi who is 7 now, and is my biggest love and pride in the world. 
                            </p>
                        </div>
                        <div>
                            <FontAwesomeIcon icon="dog"/>
                            <p>
                                We have a lovely little Yorkshire Terrier named Belle, who is a genuine part of our family, and who never 
                                spends a night outside of our bed.
                            </p>
                        </div>

                         <Image src={programmingPic} roundedCircle className="about-profile-pic" />
                         <h2>Me as a software engineer</h2>
                         <p>
                             I made quite a few turn arounds during my proffessional career. First I was in love with image processing. 
                             Then with elecro-optics, specifically with optical design, but now I feel that I found what I was looking 
                             for all this time - software. 
                         </p>
                         <p>
                             Having the experience and a broader look of the makes me a better programmer. I guess that it's called maturity,
                             and I found, that it is a required asset in our industry.
                         </p>
                         <p>
                             I really am into good coding practices. I watched many lectures of SW world icons as 'Uncle Bob', Martin Fowler,
                             Linus Torvalds, Guido Van-Rossum, Kevlin Henney, Raymond Hettinger and others.
                             I also have read several books regarding this subject, amongs them 'Clean Code' by Bob Martin and 'Refactoring' by Martin Fowler.
                         </p>
                         <p>
                             I understand that learning is not enough. In order to become profficient and comfortable with new and existing technologies,
                             you have to practice, and practice a lot. The same as we expect our pilots, phisicians and repair-men to be experts, so must we.
                             I found that except practicing and getting better all the time in our daily work, there are other great platforms like
                             excersism.io, freeCodeCamp, Medium, youtube and others.
                         </p>
                         <p>
                             One thing that I'm still postponing, hopefully not for long, is taking part in an open source code project.
                             It should be really awesome, and I can't wait to get to it rather sooner than later.
                         </p>


                         <h2>Check us out...</h2>
                         <ImageGallery />
                    </Col>
                </Container>               
            </div>
        );
    }
}