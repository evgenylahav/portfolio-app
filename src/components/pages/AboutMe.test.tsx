import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { AboutMe } from './AboutMe';
import { shallow } from 'enzyme';
import { testNameToKey } from 'jest-snapshot/build/utils';


describe('AboutMe unit test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<AboutMe />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('checks that only one about me page is created', () => {
    const element = shallow(
      <AboutMe />
    );
    expect(element.find(`[data-test='aboutme-div']`).length).toBe(1);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(
      <AboutMe />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
