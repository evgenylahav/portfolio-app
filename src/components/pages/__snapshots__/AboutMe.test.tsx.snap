// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMe unit test has a valid snapshot 1`] = `
<div
  className="aboutme-grid"
  data-test="aboutme-div"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-8 col-12"
        style={
          Object {
            "paddingBottom": "100px",
          }
        }
      >
        <h3>
          A few facts about myself
        </h3>
        <ul>
          <li>
            I love traveling. Especially with my family. We have explored several unforgetable places and hopefully many more will follow.
          </li>
          <li>
            I am a sports enthusiast, although, unfortunatelly, as a viewer. I love Tennis and Basketball the most, but during big events such as the Olympics or the World Cup, I watch everything there is.
          </li>
          <li>
            I love food. Both making and eating. This is really a weakness of mine, and I'm paying for that with a horrible BMI rate.
          </li>
          <li>
            I love coffee - can't live without it. I usually drink 5-6 cups a day, with a mixture of cappucinno, black Turkish and my favorite - double espresso.
          </li>
          <li>
            I am a fan of TV shows, watching quite a few on Netflix and other platforms. Among the best I've watched are: 'Better Call Soul', 'Dark', 'Bron', 'Breaking Bad', 'Fargo', 'Narcos' and others
          </li>
        </ul>
      </div>
      <div
        className="col-sm-4 col-12"
      >
        <img
          className=""
          src="romi.jpeg"
          style={
            Object {
              "marginLeft": "50px",
              "maxHeight": "200px",
              "width": "auto",
            }
          }
        />
        <hr />
        <img
          className=""
          src="family.jpeg"
          style={
            Object {
              "marginLeft": "50px",
              "maxHeight": "200px",
              "width": "auto",
            }
          }
        />
        <hr />
      </div>
      <div
        className="col-sm-8 col-12"
      >
        <h3>
          My family
        </h3>
        <ul>
          <li>
            I am happily married to Yardena who I love very much. She is a school councelor and a pschycoterapist, and one of the smartest people I know. We are best friends for over a decade and a half, and she definitely is the most influencial person in my life.
          </li>
          <li>
            I am a father of Romi who is 7 now, and is my biggest love and pride in the world. She is a smart and beautiful girl, who likes to swim, play the piano and watch tv.
          </li>
          <li>
            We have a lovely little Yorkshire Terrier named Belle, who is a genuine part of our family, and who never spends a night outside of our bed.
          </li>
        </ul>
      </div>
      <div
        className="col-sm-4 col-12"
      >
        <img
          className=""
          src="kodon_solo.jpeg"
          style={
            Object {
              "marginLeft": "50px",
              "maxHeight": "200px",
              "width": "auto",
            }
          }
        />
        <hr />
        <img
          className=""
          src="romi_bw.jpeg"
          style={
            Object {
              "marginLeft": "50px",
              "maxHeight": "200px",
              "width": "auto",
            }
          }
        />
        <hr />
      </div>
      <h3>
        Me as a software engineer
      </h3>
      <div
        className="col-sm-8 col-12"
        style={
          Object {
            "paddingBottom": "100px",
          }
        }
      >
        <ul>
          <li>
            I made quite a few turn arounds during my proffessional career. First I was in love with image processing. Then with elecro-optics, specifically with optical design, but now I feel that I found what I was looking for all this time - software.
          </li>
          <li>
            Having the experience and a broader look of the makes me a better programmer. I guess that it's called maturity, and I found, that it is a required asset in our industry.
          </li>
          <li>
            I really am into good coding practices. I watched many lectures of SW world icons as 'Uncle Bob', Martin Fowler, Linus Torvalds, Guido Van-Rossum, Kevlin Henney, Raymond Hettinger and others. I also have read several books regarding this subject, amongs them 'Clean Code' by Bob Martin and 'Refactoring' by Martin Fowler.
          </li>
          <li>
            I understand that learning is not enough. In order to become profficient and comfortable with new and existing technologies, you have to practice, and practice a lot. The same as we expect our pilots, phisicians and repair-men to be experts, so must we. I found that except practicing and getting better all the time in our daily work, there are other great platforms like excersism.io, freeCodeCamp, Medium, youtube and others.
          </li>
          <li>
            One thing that I'm still postponing, hopefully not for long, is taking part in an open source code project. It should be really awesome, and I can't wait to get to it rather sooner than later.
          </li>
        </ul>
      </div>
      <div
        className="col-sm-4 col-12"
      >
        <img
          className=""
          src="romi_kodon.jpeg"
          style={
            Object {
              "marginLeft": "50px",
              "maxHeight": "200px",
              "width": "auto",
            }
          }
        />
        <hr />
        <img
          className=""
          src="romi_with_model.jpeg"
          style={
            Object {
              "marginLeft": "50px",
              "maxHeight": "200px",
              "width": "auto",
            }
          }
        />
        <hr />
      </div>
      <div
        className="col-sm-8 col-12"
        style={
          Object {
            "paddingBottom": "100px",
          }
        }
      >
        <h3>
          Recommended Books
        </h3>
        <p>
          Since we are already into speaking about programming, here are a few great books that I have read and can recommend.
        </p>
        <p>
          Disclosure 1: I'm not getting anything out of it.
        </p>
        <p>
          Disclosure 2: I didn't read all the content of all of these books. Sometimes I jump back and forth, depending of my current interests.
        </p>
        <ul>
          <li>
            <a
              href="https://roadtoreact.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              The road to learn React |
            </a>
            <a
              href="https://twitter.com/rwieruch"
              rel="noopener noreferrer"
              target="_blank"
            >
              | Robin Wieruch
            </a>
          </li>
          <li>
            <a
              href="https://martinfowler.com/books/refactoring.html"
              rel="noopener noreferrer"
              target="_blank"
            >
              Refactoring |
            </a>
            <a
              href="https://twitter.com/martinfowler"
              rel="noopener noreferrer"
              target="_blank"
            >
              | Martin Fowler
            </a>
          </li>
          <li>
            <a
              href="https://www.amazon.com/Clean-Code-Handbook-Software-Craftsmanship/dp/0132350882"
              rel="noopener noreferrer"
              target="_blank"
            >
              Clean Code |
            </a>
            <a
              href="https://twitter.com/unclebobmartin"
              rel="noopener noreferrer"
              target="_blank"
            >
              | Bob Martin (Uncle Bob)
            </a>
          </li>
          <li>
            <a
              href="https://effectivepython.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Effective Python |
            </a>
            <a
              href="https://twitter.com/haxor"
              rel="noopener noreferrer"
              target="_blank"
            >
              | Brett Slatkin
            </a>
          </li>
          <li>
            <a
              href="http://effectivejs.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              Effective JavaScript |
            </a>
            <a
              href="https://twitter.com/littlecalculist"
              rel="noopener noreferrer"
              target="_blank"
            >
              | David Herman
            </a>
          </li>
        </ul>
      </div>
      <div
        className="col-sm-4 col-12"
      >
        <img
          className=""
          src="romi_an_belle.jpeg"
          style={
            Object {
              "marginLeft": "110px",
              "maxHeight": "200px",
              "width": "auto",
            }
          }
        />
        <hr />
        <img
          className=""
          src="belle.jpeg"
          style={
            Object {
              "marginLeft": "50px",
              "maxHeight": "200px",
              "width": "auto",
            }
          }
        />
        <hr />
      </div>
    </div>
  </div>
</div>
`;
